aiida/cmdline/commands/profile.py

```diff
@@ -29,162 +31,144 @@ def __init__(self):
+        :param subargs_idx: flag
+        :param subargs: additional arguments
+
+        :return: either profiles list or None.
+        """
         from aiida.common.setup import get_profiles_list
 
         if subargs_idx == 0:
             return "\n".join(get_profiles_list())
+
+        return ""
+
+
+@verdi_profile.command("list")
+def profile_list():
+    """
+    Displays list of all available profiles.
+    """
+
+    from aiida.common.setup import get_profiles_list, get_default_profile, AIIDA_CONFIG_FOLDER
+    from aiida.common.exceptions import ConfigurationError
+
+    echo.echo_info('Configuration folder: {}'.format(AIIDA_CONFIG_FOLDER))
+
+    try:
+        default_profile = get_default_profile()
+    except ConfigurationError as err:
+        err_msg = ("Stopping: {}\n"
+                   "Note: if no configuration file was found, it means that you have not run\n"
+                   "'verdi setup' yet to configure at least one AiiDA profile.".format(err.message))
+        echo.echo_critical(err_msg)
+
+    if default_profile is None:
+        echo.echo_critical("### No default profile configured yet, run 'verdi install'! ###")
+    else:
+        echo.echo_info('The default profile is highlighted and marked by the * symbol')
+
+    for profile in get_profiles_list():
+        color_id = ''
+        if profile == default_profile:
+            symbol = '*'
+            color_id = 'green'
         else:
-            return ""
+            symbol = ' '
 
-    def profile_setdefault(self, *args):
-        from aiida.common.setup import set_default_profile
+        click.secho('{} {}'.format(symbol, profile), fg=color_id)
 
-        if len(args) != 1:
-            print >> sys.stderr, ("Please specify the profile to be set as the default")
-            sys.exit(1)
 
-        profile = args[0]
-        # set default DB profiles
-        set_default_profile(profile, force_rewrite=True)
+@verdi_profile.command("setdefault")
+@click.argument('profile_name', nargs=1, type=click.STRING)
+def profile_setdefault(profile_name):
+    """
+    Set the passed profile as default profile in aiida config file.
+    """
+    from aiida.common.setup import set_default_profile
+    set_default_profile(profile_name, force_rewrite=True)
 
 
-    def profile_list(self, *args):
-        from aiida.backends import settings
-        from aiida.common.setup import get_profiles_list, get_default_profile, AIIDA_CONFIG_FOLDER
-        from aiida.common.exceptions import ConfigurationError
+@verdi_profile.command("delete")
+@options.FORCE(help='to skip any questions/warnings about loss of data')
+@click.argument('profiles_to_delete', nargs=-1, type=click.STRING)
+def profile_delete(force, profiles_to_delete):
+    """
+    Delete profile from aiida config file along with its associated
+    database and repository.
 
-        print('Configuration folder: {}'.format(AIIDA_CONFIG_FOLDER))
+    :param profiles_to_delete: list of profiles separated by space
```

Comment:
Do not use explicit `:param:` notation in the docstring. You can define `metavar=PROFILES` in the `click.argument` and use that name in the docstring e.g.:

```
Delete PROFILES from the aiida configuration file along with
their associated database and repository.
```

Reply:
This is because in `click`, the docstring of a command becomes it's help text